Sources/ProjectSpec/TargetSource.swift

```diff
@@ -181,6 +181,32 @@ extension TargetSource: JSONObjectConvertible {
     }
 }
 
+extension TargetSource: JSONEncodable {
+    public func toJSONValue() -> Any {
+        var dict: [String: Any?] = [
+            "compilerFlags": compilerFlags,
+            "excludes": excludes,
+            "name": name,
+            "headerVisibility": headerVisibility?.rawValue,
+            "type": type?.rawValue,
+            "buildPhase": buildPhase?.toJSONValue(),
+            "createIntermediateGroups": createIntermediateGroups,
+        ]
+
+        if optional {
+            dict["optional"] = true
```

Comment:
I'm a bit worried about the maintainability of all these properties, and that regressions might occur in the future by accident. 

To aid with the properties that have defaults, and because we're now mapping those defaults in 3 places (property init, decoding, and now encoding), it would be good if we have static properties for all of these. In this example:
```
static let defaultOptional = true
```

Reply:
@yonaskolb It's make sense. Since I changed it, could you please review it?

Reply:
Looks good! Could you please rebase to fix the merge conflict and add a changelog?